framework/db/Command.php

```diff
@@ -1140,8 +1140,7 @@ protected function queryInternal($method, $fetchMode = null)
             if (is_array($info)) {
                 /* @var $cache \yii\caching\CacheInterface */
                 $cache = $info[0];
-                $rawSql = $rawSql ?: $this->getRawSql();
-                $cacheKey = $this->getCacheKey($method, $fetchMode, $rawSql);
+                $cacheKey = $this->getCacheKey($method, $fetchMode);
```

Comment:
I think the line `$rawSql = $rawSql ?: $this->getRawSql();` shouldn't be removed, because if `$rawSql` is not empty we no need to recalculate it again.

Reply:
getRawSql is the place which cause db connection (quoteValue), but you are right, if there is no cache, rawsql maybe empty, and the code after cache may cause error

Reply:
> getRawSql is the place which cause db connection (quoteValue), but you are right, if there is no cache, rawsql maybe empty, and the code after cache may cause error

after research the code, rawsql is only used for log, if profile enabled it will be set, so there is no problem